lib/device-log/safari-network-log.js

```diff
@@ -0,0 +1,140 @@
+import _ from 'lodash';
+import URL from 'url';
+import { RotatingLog, MAX_LOG_ENTRIES_COUNT } from './rotating-log';
+
+
+class SafariNetworkLog extends RotatingLog {
+  constructor (showLogs) {
+    super(showLogs, 'SafariNetwork');
+  }
+
+  getEntry (requestId) {
+    let outputEntry;
+    while (this.logs.length >= MAX_LOG_ENTRIES_COUNT) {
+      // pull the first entry, which is the oldest
+      const entry = this.logs.shift();
+      if (entry && entry.requestId === requestId) {
+        // we are adding to an existing entry, and it was almost removed
+        // add to the end of the list and try again
+        outputEntry = entry;
+        this.logs.push(outputEntry);
+        continue;
+      }
+      // we've removed an element, so the count is down one
+      if (this.logIdxSinceLastRequest > 0) {
+        this.logIdxSinceLastRequest--;
+      }
+    }
+
+
+    if (!outputEntry) {
+      // we do not yes have an entry to associate this bit of output with
+      // most likely the entry will be at the end of the list, so start there
+      for (let i = this.logs.length - 1; i >= 0; i--) {
```

Comment:
you could use `for item of _.reverse(arr)` instead

Reply:
If the logs length is long, this would be _very_ inefficient.

Reply:
yep, keep forgetting it is not a generator method